batcher/aligned-batcher/src/s3/mod.rs

```diff
@@ -5,13 +5,22 @@ use aws_sdk_s3::operation::put_object::{PutObjectError, PutObjectOutput};
 use aws_sdk_s3::primitives::ByteStream;
 use aws_sdk_s3::Client;
 
-pub async fn create_client() -> Client {
+pub async fn create_client(endpoint_url: Option<String>) -> Client {
     let region_provider = RegionProviderChain::default_provider().or_else("us-east-2");
-    let config = aws_config::defaults(BehaviorVersion::latest())
-        .region(region_provider)
+    let mut config = aws_config::defaults(BehaviorVersion::latest())
+        .region(region_provider);
+    if let Some(endpoint_url) = &endpoint_url {
+        config = config.endpoint_url(endpoint_url);
+    }
+    let config = config
         .load()
         .await;
-    Client::new(&config)
+
+    let mut s3_config_builder = aws_sdk_s3::config::Builder::from(&config);
+    if Some(endpoint_url).is_some() {
+        s3_config_builder = s3_config_builder.force_path_style(true);
+    }
```

Comment:
this will always return true hehe :). What are you checking here? If you want to check if `endpoint_url` is `Some`, you already did in line 12.